src/MMH3.cc

```diff
@@ -21,47 +21,40 @@ uint32_t detail::MMH3Val::Result() {
   return PMurHash32_Result(mmh3_h1, mmh3_carry, mmh3_total_len);
 }
 
-broker::expected<broker::data> detail::MMH3Val::DoSerialize() const {
+std::optional<zeek::BrokerData> detail::MMH3Val::DoSerializeData() const {
   if (!IsValid())
-    return {broker::vector{false}};
+    return std::nullopt;
 
-  broker::vector d = {true, mmh3_h1, mmh3_carry, mmh3_total_len};
-  return {std::move(d)};
+  zeek::BrokerListBuilder builder;
+  builder.Reserve(4);
+  builder.Add(true);
+  builder.Add(static_cast<uint64_t>(mmh3_h1));
+  builder.Add(static_cast<uint64_t>(mmh3_carry));
+  builder.Add(static_cast<uint64_t>(mmh3_total_len));
+
+  return std::move(builder).Build();
 }
 
 // This is a totally untested best-effort implementation. It's a
 // bit of a burden adding this to a HashVal.
-bool detail::MMH3Val::DoUnserialize(const broker::data &data) {
-  auto *d = broker::get_if<broker::vector>(data);
-  if (!d)
+bool detail::MMH3Val::DoUnserializeData(zeek::BrokerDataView data) {
+  if (!data.IsList())
     return false;
 
-  const auto &dv = *d;
-
-  if (dv.size() < 1)
-    return false;
+  auto d = data.ToList();
 
-  auto *valid = broker::get_if<broker::boolean>(dv[0]);
-  if (!valid)
+  if (d.Size() != 4 || !d[0].IsBool())
     return false;
 
-  if (!*valid)
-    return true; // while the content is invalid, unserializing was
-                 // successful.
-
-  if (dv.size() < 4)
+  if (!d[1].IsCount() || !d[2].IsCount() || !d[3].IsCount())
     return false;
```

Comment:
Nit: could probably clean up checking for 1) number of values, and 2) value types, e.g.,

```suggestion
  if (d.Size() != 4)
    return false;

  if (!(d[0].IsBool() && d[1].IsCount() && d[2].IsCount() && d[3].IsCount()))
    return false;
```